src/nifs/vanilla/mod.rs

```diff
@@ -339,25 +342,57 @@ impl<C: CurveAffine> VerifyAccumulation<C> for VanillaFS<C> {
         S: &PlonkStructure<C::ScalarExt>,
         acc: &<Self as FoldingScheme<C>>::Accumulator,
     ) -> Result<(), Self::VerifyError> {
+        /// Under this collapsing scheme, `instance` columns other than consistency markers are not
+        /// foldeded, but accumulated using hash. Therefore, they need to be cut out for
+        /// `is_sat_permutation`.
+        ///
+        /// To account for these permutations in the `Relaxed` version, we add them to StepFoldingCircuit
+        /// as a copy constraint with private input (witness)
+        fn permutation_data_without_step_circuit_instances<F: PrimeField>(
+            S: &PlonkStructure<F>,
+        ) -> SparseMatrix<F> {
+            S.permutation_data
+                .clone()
+                .rm_copy_constraints(1..S.num_io.len())
```

Comment:
there are two approaches. let's say the cycles before is (2,3)->(1,2)->(4,5)->(2,3), and we want to remove column=1.
(1) the approach in this PR: change it to (2,3)->(4,5)->(2,3) and (1,2)->(1,2)
(2) Another slightly better approach: change it to : (2,3)->(4,5)->(2,3) and completely remove (1,2) from both column set and mappings.

For the approach (2), we can pre-process the permutation data to remove them when creating S.  Since when we concatenate instance data with witness, we already removed the columns to be removed, the checking of `is_sat_permutation` will pass.

Reply:
At the end of the comment you described the implemented approach, except that I perform the deletion copy-constraints already at the stage of calling the folding scheme verification.

That is, we store normal data in `PlonkStructure` and modify it only when necessary

Reply:
We concatenate instance & witness already in the NIFS module. I.e. there is no problem to remove these columns with current approach, I just decided that keeping the original dimensionality is easier and debugging reading this code is easier.